hw/opentitan/ot_spi_device.c

```diff
@@ -2446,21 +2449,36 @@ static void ot_spi_device_chr_recv_flash(OtSPIDeviceState *s,
     }
 }
 
-static bool ot_spi_device_tpm_is_hw_register(uint32_t addr)
-{
-    static const uint32_t TPM_HW_ADDR[] = {
-        0x000u, /* 000     Access_x*/
-        0x008u, /* 00B:008 Interrupt Enable*/
-        0x00Cu, /* 00C     Interrupt Vector*/
-        0x010u, /* 013:010 Interrupt Status*/
-        0x014u, /* 017:014 Interface Capability*/
-        0x018u, /* 01B:018 Status_x*/
-        0x028u, /* 028     Hash Start*/
-        0xF00u, /* F03:F00 DID_VID*/
-        0xF04u /* F04:F04 RID*/
+static bool
+ot_spi_device_tpm_get_hw_register(OtSPIDeviceState *s, uint32_t *reg_val)
+{
+    static const TpmHwAddrMap TPM_HW_ADDR[] = {
+        /*{ register_addr, tpm_register_offset, size }*/
+        { 0x000u, R_TPM_ACCESS_0, 4u }, /* 000     Access_x*/
+        { 0x004u, R_TPM_ACCESS_1, 4u }, /* 000     Access_x*/
+        { 0x008u, R_TPM_INT_ENABLE, 4u }, /* 00B:008 Interrupt Enable*/
+        { 0x00Cu, R_TPM_INT_VECTOR, 1u }, /* 00C     Interrupt Vector*/
+        { 0x010u, R_TPM_INT_STATUS, 4u }, /* 013:010 Interrupt Status*/
+        { 0x014u, R_TPM_INTF_CAPABILITY, 4u }, /* 017:014 Interface Capability*/
+        { 0x018u, R_TPM_STS, 4u }, /* 01B:018 Status_x*/
+        { 0x028u, 0u, 0u }, /* 028     Hash Start*/
+        { 0xF00u, R_TPM_DID_VID, 4u }, /* F03:F00 DID_VID*/
+        { 0xF04u, R_TPM_RID, 4u }, /* F04:F04 RID*/
     };
+
+    if (reg_val) {
+        *reg_val = 0xffffffffu;
```

Comment:
Nit: QEMU style would prefer
```suggestion
        *reg_val = UINT32_MAX;
```